llpc/translator/lib/SPIRV/SPIRVReader.cpp

```diff
@@ -6336,6 +6336,32 @@ Value *SPIRVToLLVM::transSPIRVImageQueryLodFromInst(SPIRVInstruction *bi, BasicB
     result = ConvertingSamplerSelectLadderHelper(result, convertingSamplerIdx, createImageGetLod);
   }
 
+  // NOTE: This is a workaround. When UV width equals 0, the result return 0, but we expect the value is
+  // -22.0 to -infinity.
+  // Derivative operations are performed on the XY of UV coordinate respectively, if both of result
+  // are zero, we will return -FLT_MAX for LOD.
+  Value *dpdx = getBuilder()->CreateDerivative(coord, false, false);
+  Value *dpdy = getBuilder()->CreateDerivative(coord, true, false);
+  Value *absDpdx = getBuilder()->CreateUnaryIntrinsic(Intrinsic::fabs, dpdx);
```

Comment:
Let's try to reduce some local variables
Value *absDpdx = getBuilder()->CreateUnaryIntrinsic(Intrinsic::fabs, getBuilder()->CreateDerivative(coord, false, false));
Value *absDpdy = ...
Remove dpdx and dpdy